internal/mocks/atlas/global_clusters.go

```diff
@@ -0,0 +1,98 @@
+package atlas
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+type GlobalClustersClientMock struct {
```

Comment:
@helderjs can you elaborate on why we need this GlobalClusters mock? Probably my issue is I do not know what GlobalClusters is all about.

It is not an objection to the PR, just I can't figure this out by myself.

Reply:
Global Cluster is a feature of the deployments(clusters) that allows users to group single or multi-regions in Zones. Those can be configured to have read/write to geographically set shards([see more here](https://www.mongodb.com/docs/atlas/global-clusters/)).
As the reconciliation calls endpoints in this resource I had to add its mock